src/components/redux/State.js

```diff
@@ -37,25 +37,28 @@ const store = {
   _callSubscriber() {
     console.log('State changed');
   },
-  addPost(){
-    let newPost = {
-      id: 5,
-      message: this._state.ProfilePage.newPostText , 
+  
+  subscribe(observer) {
+    this._callSubscriber = observer;
+  } ,
 
-      likesCount: 0
-    };
-    this._state.ProfilePage.postsData.push(newPost);
-    this._state.ProfilePage.newPostText = ('');
-    this._callSubscriber(this._state);
-  },
-  updateNewpostText(newText) {
-    this._state.ProfilePage.newPostText = newText;
+  dispatch(action)  {
+    if (action.type === 'ADD-POST') {
```

Comment:
Consider creating a file named "actionTypes.js" and storing there an object with all the action names.
`const ACTION_TYPES = { ..., ADD_POST: 'ADD_POST', ...}`;
This way you can import this object everywhere you need and reference needed type like this
`action.type === ACTION_TYPES.ADD_POST`;

Reply:
done